File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/CreatePayoutCommand.java
package com.hedvig.paymentservice.domain.payments.commands;

import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;
import javax.money.MonetaryAmount;

import com.hedvig.paymentservice.domain.payments.TransactionCategory;
import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

@Value
public class CreatePayoutCommand {
  @TargetAggregateIdentifier String memberId;
  String address;
  String countryCode;
  LocalDate dateOfBirth;
  String firstName;
  String lastName;
  UUID transactionId;
  MonetaryAmount amount;
  Instant timestamp;
  TransactionCategory category;
  String referenceId;
  String note;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/CreatePayoutCommand.kt
package com.hedvig.paymentservice.domain.payments.commands

import com.hedvig.paymentservice.domain.payments.TransactionCategory
import org.axonframework.commandhandling.TargetAggregateIdentifier
import java.time.Instant
import java.time.LocalDate
import java.util.UUID
import javax.money.MonetaryAmount

data class CreatePayoutCommand(
  @TargetAggregateIdentifier
  val memberId: String,
  val address: String?,
  val countryCode: String?,
  val dateOfBirth: LocalDate?,
  val firstName: String?,
  val lastName: String?,
  val transactionId: UUID,
  val amount: MonetaryAmount,
  val timestamp: Instant,
  val category: TransactionCategory,
  val referenceId: String?,
  val note: String?,
  val handler: String?,
  val email: String
)
